scripts/readscount.py

```diff
@@ -7,212 +7,271 @@
         f.write("sep=,\n")
         f.write(
             "sample name,number of flowcells,number of lanes,flowcell1:lane1|lane2;flowcell2:lane1|lane2|lane3 ...\n"
         )
         for sample in summary:
             view = []
-            totfc = len(summary[sample])
-            totlanes = 0
+            n_flowcells = len(summary[sample])
+            n_lanes = 0
             for fc in summary[sample]:
                 view.append("{}:{}".format(fc, "|".join(summary[sample][fc])))
-                totlanes += len(summary[sample][fc])
-            f.write("{},{},{},{}\n".format(sample, totfc, totlanes, ";".join(view)))
+                n_lanes += len(summary[sample][fc])
+            f.write(
+                "{},{},{},{}\n".format(sample, n_flowcells, n_lanes, ";".join(view))
+            )
+
+    # Upload the log file
     try:
-        attach_file(os.path.join(os.getcwd(), "AggregationLog.csv"), logart)
-        logging.info(f"updated {samplenb} samples with {errnb} errors")
+        attach_file(os.path.join(os.getcwd(), "AggregationSummary.csv"), log_artifact)
+        logging.info(f"Updated {sample_counter} samples with {error_counter} errors")
     except AttributeError:
-        # happens if the log artifact does not exist, if the step has been started before the configuration changes
+        # Happens if the log artifact does not exist, if the step has been started before the configuration changes
         logging.info("Could not upload the log file")
 
 
-def demnumber(sample):
-    """Returns the number of distinct demultiplexing processes tagged with "Include reads" for a given sample"""
-    expectedName = f"{sample.name} (FASTQ reads)"
-    dem = set()
-    arts = lims.get_artifacts(
-        sample_name=sample.name,
-        process_type=list(DEMULTIPLEX.values()),
-        name=expectedName,
-    )
-    for a in arts:
-        if a.udf["Include reads"] == "YES":
-            dem.add(a.parent_process.id)
-    return len(dem)
+def sum_reads(sample, summary):
+    """For a given submitted sample and it's summary,
+    calculate the total number of reads and append to the summary.
+    """
 
+    logging.info(f"Aggregating reads of sample '{sample.name}'...")
 
-def sumreads(sample, summary):
+    # Append to summary
     if sample.name not in summary:
         summary[sample.name] = {}
-    expectedName = f"{sample.name} (FASTQ reads)"
-    arts = lims.get_artifacts(
+
+    # Look for artifacts matching the sample name and expected analyte name in the demultiplexing processeses
+    demux_arts = lims.get_artifacts(
         sample_name=sample.name,
         process_type=list(DEMULTIPLEX.values()),
-        name=expectedName,
+        name=f"{sample.name} (FASTQ reads)",
     )
-    tot = 0
-    fclanel = []
-    filteredarts = []
-    base_art = None
-    for a in sorted(arts, key=lambda art: art.parent_process.date_run, reverse=True):
-        if "# Reads" not in a.udf:
+    if not demux_arts:
+        raise AssertionError(
```

Comment:
Couldn't this happen if only some samples have been sequenced? A bit harsh to raise an exception that will abort the entire script?

Reply:
Yes, I assumed one wouldn't want to run the script until all sequencing was finished, but I could replace this with a red banner and a warning, if preferable.

Reply:
[d0addab](https://github.com/NationalGenomicsInfrastructure/scilifelab_epps/pull/386/commits/d0addab65e0e5062bacee1a37f65ffcfebec675c)